packages/vite/src/node/utils.ts

```diff
@@ -1028,16 +1028,20 @@ function normalizeSingleAlias({
   return alias
 }
 
-export function transformResult(
+/**
+ * Transforms transpiled code result where line numbers aren't altered,
+ * so we can skip sourcemap generation during dev
+ */
+export function transformStableResult(
   s: MagicString,
   id: string,
   config: ResolvedConfig
 ): TransformResult {
-  const isBuild = config.command === 'build'
-  const needSourceMap = !isBuild || config.build.sourcemap
   return {
     code: s.toString(),
-    map: needSourceMap ? s.generateMap({ hires: true, source: id }) : null
+    map: config.build.sourcemap
```

Comment:
if config `config.build.sourcemap` dev is slow too?

Reply:
Good catch, actually we should opt-out during dev here too. It is false by default, so most people weren't affected. But this should ignore the maps during dev also when this flag is on.

Reply:
done in [fix: only use build.sourcemap during build](https://github.com/vitejs/vite/pull/8796/commits/7aa6e58a29192ceaf1fdd8b120e607214a51bec1)